### ISSUE TITLE
Unexpected `$each` operator is being added to array updates

### ISSUE THREAD

**Issue by letsgolesco** [Automattic/mongoose#6941]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Using `$addToSet` to add a string array to an array, mongoose appends `$each` to the update so each string within the string array is separated into its own array.

I think the fix for this issue is what's causing my issue: https://github.com/Automattic/mongoose/issues/6086

Example script to reproduce:
```javascript
const exampleSchema = {
  exampleField: [[String]]
};
const Example = mongoose.schema('Example', ExampleSchema);
Example.create({ exampleField: [['A', 'B']] })
.then(createdDoc => {
  const query = { _id: createdDoc._id };
  const update = { $addToSet: { exampleField: ['C', 'D'] } };
  const options = { new: true, runValidators: true };
  Model.findOneAndUpdate(query, update, options)
    .exec()
    .then(updatedDoc => console.log(updatedDoc);
  // updatedDoc = { exampleField: [['A', 'B'], ['C'], ['D']] }
  // expected = { exampleField: [['A', 'B'], ['C', 'D']] } 
});
```

**What is the expected behavior?**
I did not expect $each to be added behind the scenes. I expected the array to be added to the set as it is in the MongoDB shell.

I've found a compromise, which is to override the `$each` operator for my desired behavior. So in the above example, my update would become:
```javascript
const update = { $addToSet: { exampleField: { $each: [['C', 'D']] } } };
```

**Please mention your node.js, mongoose and MongoDB version.**
Node: 8.11.4
Mongoose: 5.2.9
MongoDB: 3.6.2

**Comment by lineus** [comment]
@letsgolesco You're correct that $each shouldn't be added when adding an array to an array. I thought that this behavior was fixed in 5.2.7. Is it possible you are running this with a slightly older version than 5.2.9, like 5.2.6, etc?

Here's roughly the same example as the one you provided running on 5.2.9 and it seems to be working correctly.

### 6941.js
```js
#!/usr/bin/env node --no-deprecation
'use strict';

const assert = require('assert');
const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost:27017/test', { useNewUrlParser: true });
const conn = mongoose.connection;
const Schema = mongoose.Schema;

const schema = new Schema({
  arr: [[String]]
});

const Test = mongoose.model('test', schema);

const test = new Test({ arr: [['a','b']]});

async function run() {
  assert.strictEqual(mongoose.version, '5.2.9');
  await conn.dropDatabase();
  await test.save();
  let cond = { _id: test._id };
  let update = { $addToSet: { arr: ['c', 'd'] } };
  let opts = { new: true };
  await Test.findOneAndUpdate(cond, update, opts);
  let updated = await Test.findOne();
  assert.deepStrictEqual(updated.toObject().arr, [['a', 'b'], ['c','d']]);
  console.log('all assertions pass');
  return conn.close();
}

run();
```
### Output:
```
issues: ./6941.js
all assertions pass
issues:
```

**Comment by vkarpov15** [comment]
Thanks for reporting, will fix asap
